src/Enum.php

```diff
@@ -205,8 +224,8 @@ protected static function resolve(): array
         }
 
         foreach (self::$cache[$class] as $name => $enum) {
```

Comment:
Could this loop be combined with the one above?

```php
foreach ($values as $index => $value) {
    $name = strtoupper($value);

    self::$cache[$class][$name] = [
        'index' => static::getMappedIndex($name) ?? static::make($name)->getIndex(),
        'value' => static::getMappedValue($name) ?? static::make($name)->getValue(),
    ];
}
```

Maybe not because of the `static::make` calls?

Reply:
> Maybe not because of the static::make calls?

right - we need something present in the `self::$cache[$class]` to resolve the custom values/index. Only the mapped value could be moved up - this would also solve the issue of conflicting map constant and custom getter.

Reply:
https://github.com/spatie/enum/pull/25/commits/d697dc49059372a8a1ecd808d9b3731981d897c2